src/main/java/frc/robot/states/ShootState.java

```diff
@@ -0,0 +1,36 @@
+package frc.robot.states;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.swerve.SwerveDrive;
+
+public class ShootState implements ScoreState {
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () -> {
+                    var optimalPoints = Constants.OPTIMAL_POINTS_SHOOT_BLUE;
+                    if (isRed()) {
+                        optimalPoints = Constants.OPTIMAL_POINTS_SHOOT_RED;
+                    }
```

Comment:
This is a nice place to use a ternary operation:
```suggestion
                    var optimalPoints = isRed() ? Constants.OPTIMAL_POINTS_SHOOT_RED : Constants.OPTIMAL_POINTS_SHOOT_BLUE;
```
Same in Trap state.

Reply:
fbc6355